### ISSUE TITLE
Position Management System

### ISSUE THREAD

**Issue by alexflorisca** [bumpy-croc/ai-trading-bot#33]
Can you give me a markdown friendky version of your last output I can post in github projects
# Position Management - Architecture & Best-Practices

> “Entries are optional. Exits are **everything**.” –  ​Anonymous CTA

---

## 1. What *is* Position Management?

Position management is the continuous process of **monitoring, adjusting, and optimising** open trades from entry to final exit.  
It controls risk, preserves capital, and squeezes the most out of profitable moves.

---

## 2. Core Components

| # | Component | Purpose |
|---|-----------|---------|
| 1 | **Position Tracking & State** | Persist everything about a trade (size, avg price, PnL, MFE/MAE, age, regime at entry…). |
| 2 | **Dynamic Risk Management** | Adjust stops/targets as market evolves, trail winners, defend losers. |
| 3 | **Exit Strategies** | Price-, time-, and indicator-based rules to close or scale positions. |
| 4 | **Position Sizing Adjustments** | Scale-in / scale-out logic, volatility sizing, Kelly/Fixed-Fractional. |
| 5 | **Advanced Stop-Losses** | ATR / volatility stops, breakeven moves, trailing algorithms. |
| 6 | **Correlation Control** | Cap exposure to highly-correlated assets or factors. |
| 7 | **Performance Feedback** | Tighten risk when strategy under-performs, loosen when it excels. |

---

### 2.1 Position Dataclass Example

```python
@dataclass
class Position:
    symbol: str
    side: PositionSide          # long / short
    size: float
    entry_price: float
    entry_time: datetime
    stop_loss: float | None = None
    take_profit: float | None = None
    unrealised_pnl: float = 0.0
    mfe: float = 0.0            # max favourable excursion
    mae: float = 0.0            # max adverse excursion
    order_id: str | None = None
```

---

### 2.2 Dynamic Risk Management Snippet

```python
def manage_position(pos: Position, md):
    # ── Trailing stop ──────────────────────────────
    if pos.unrealised_pnl > 0.015:                       # +1.5 %
        new_stop = md.price * 0.995                      # 0.5 % trail
        pos.stop_loss = max(pos.stop_loss or 0, new_stop)

    # ── Partial exits ────────────────────────────────
    if pos.unrealised_pnl > 0.03:                        # +3 %
        exit_qty = pos.size * 0.33
        broker.exit_partial(pos, qty=exit_qty, reason="First target")

    # ── Scale-in ─────────────────────────────────────
    if md.trend_strength > 0.7 and pos.unrealised_pnl > 0:
        add_size = pos.size * 0.5
        broker.add_to_position(pos, size=add_size)
```

---

### 2.3 Exit Strategy Palette

| Type | Examples |
|------|----------|
| **Time-based** | End-of-day, max hold hours, weekend flat. |
| **Price-based** | Fixed %, Fibonacci extensions, support/resistance. |
| **Indicator-based** | MA crossover, RSI divergence, volatility spike. |

---

### 2.4 Advanced Stop-Loss Techniques

```python
# Volatility-adjusted (ATR) stop
atr_mult = 2
pos.stop_loss = pos.entry_price - atr * atr_mult   # long
```

```python
# Trailing algorithm
trail_trigger = 0.015     # activate at +1.5 %
trail_dist    = 0.005     # trail 0.5 %
```

---

## 3. Integration with the Prediction Engine

```python
pred = predictor.generate_predictions(market_data)

if pred["1_bar"].confidence > 0.8 and pred["1_bar"].direction != pos.side:
    tighten_stop(pos)            # high-conf reversal

if pred["5_bar"].confidence > 0.7 and pred["5_bar"].direction == pos.side:
    maybe_scale_in(pos)          # trend continuation
```

---

## 4. Best Practices Checklist ✅

1. **Never move stops against the trade.**  
2. Use **multiple time-frames** for exits (micro + macro).  
3. Track **MFE/MAE** – vital for system evaluation.  
4. Cap **correlated exposure** (e.g., BTC & ETH).  
5. Automate; avoid discretionary tinkering.  
6. Re-optimise rules when market regimes shift.  

---

## 5. Implementation Roadmap

1. **Phase 1** – Trailing stops & partial exits.  
2. **Phase 2** – Volatility-based sizing / ATR stops.  
3. **Phase 3** – Predictor-driven dynamic management.  
4. **Phase 4** – Correlation & portfolio-level limits.  
5. **Phase 5** – ML-optimised exit timing / reinforcement learning.

---

> 💡 **Key takeaway:**  
> Entries create ideas, **position management** converts them into risk-adjusted returns.  
> Invest more dev hours here than on signal generation—you’ll thank yourself later.
